mysql57 - Fails initialize, missing persist

**### Running `scoop install mysql57` returns following error:**

```
cp : Cannot find path 'C:\Users\Zgjim Gjonbalaj\scoop\apps\mysql57\current\my-default.ini' because it does not exist.
At line:5 char:1
+ cp $dir/my-default.ini $dir/my.ini
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : ObjectNotFound: (C:\Users\Zgjim ...\my-default.ini:String) [Copy-Item], ItemNotFoundException
    + FullyQualifiedErrorId : PathNotFound,Microsoft.PowerShell.Commands.CopyItemCommand
```

Editing my.ini to include [mysqld] with a datadir and few other options also does not work. The result of running `mysqld` is:

```
mysqld: [ERROR] Found option without preceding group in config file C:\Users\Zgjim Gjonbalaj\scoop\apps\mysql57\current\my.ini at line 1!
mysqld: [ERROR] Fatal error in defaults handling. Program aborted!
```

Contents of my.ini

```
[mysqld]
innodb_buffer_pool_size = 512M
datadir                 = 'C:\Users\Zgjim Gjonbalaj\scoop\persist\mysql\data'
port                    = 3306
server_id               = 0
bind-address            = 127.0.0.1
join_buffer_size        = 512M
sort_buffer_size        = 100M
read_rnd_buffer_size    = 100M 

[client]
user                    = root
password                = root
```

It seems the original archive may or may not have the my-default.ini and the cp is failing, still does not explain the error after editing the file though i think it may have something to do with file endings/encoding.

OS: 
Win10 Enterprise Version 1803 (OS Build 17134.407)

Current Scoop version:
9a91193b (HEAD -> master, origin/master, origin/HEAD) vim: Update to version 8.1.0560

'games' bucket:
27a5bf2 (HEAD -> master, origin/master, origin/HEAD) Azure: Add `extras` bucket in test job as required by some manifests

'versions' bucket:
f634638 (HEAD -> master, origin/master, origin/HEAD) springboot15: Update to version 1.5.18

--- a/bucket/mysql56.json
+++ b/bucket/mysql56.json
@@ -1,43 +1,55 @@
 {
-    "homepage": "https://dev.mysql.com/downloads/mysql/",
+    "homepage": "https://dev.mysql.com/downloads/mysql/5.6.html",
+    "description": "The world's most popular open-source database, supported by an active community.",
     "version": "5.6.47",
-    "license": "GPLv2",
+    "license": "GPL-2.0-only",
     "architecture": {
         "64bit": {
             "url": "https://dev.mysql.com/get/mysql-5.6.47-winx64.zip",
-            "hash": "d0ef44e92669b60e22f1e8653cd115852680bd9d8be1e6073ad2bb2446dcc924",
+            "hash": "md5:a8b2634df8e088e0bb68bf607585fbb7",
             "extract_dir": "mysql-5.6.47-winx64"
         },
         "32bit": {
             "url": "https://dev.mysql.com/get/mysql-5.6.47-win32.zip",
-            "hash": "f0177205f9f6587737534ca9731df00d51ddeaaaa32f723b098ef5b51c27c14c",
+            "hash": "md5:8e57bb79e40d78e66f172ee0155b3c46",
             "extract_dir": "mysql-5.6.47-win32"
         }
     },
     "bin": [
-        "bin\\mysqld.exe",
+        "bin\\innochecksum.exe",
+        "bin\\myisamchk.exe",
+        "bin\\myisamlog.exe",
+        "bin\\myisampack.exe",
+        "bin\\myisam_ftdump.exe",
         "bin\\mysql.exe",
-        "bin\\mysqldump.exe",
         "bin\\mysqladmin.exe",
         "bin\\mysqlbinlog.exe",
         "bin\\mysqlcheck.exe",
+        "bin\\mysqld.exe",
+        "bin\\mysqldump.exe",
         "bin\\mysqlimport.exe",
         "bin\\mysqlshow.exe",
         "bin\\mysqlslap.exe",
+        "bin\\mysql_config_editor.exe",
+        "bin\\mysql_tzinfo_to_sql.exe",
+        "bin\\mysql_upgrade.exe",
         "bin\\my_print_defaults.exe"
     ],
-    "post_install": [
-        "#Initialize data directory (without generating root password)",
-        "mysqld --initialize-insecure",
-        "",
-        "#Copy provided sample file to live file location",
-        "cp $dir/my-default.ini $dir/my.ini",
-        "",
-        "#Output client configuration to my.ini file so no username is required when connecting",
-        "echo \"\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append",
-        "echo \"[client]\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append",
-        "echo \"user=root\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append"
+    "persist": [
+        "data",
+        "my.ini"
     ],
+    "pre_install": [
+        "$ini = \"$dir\\my.ini\"",
+        "if (!(Test-Path \"$ini\")) {",
+        "   # Using $dir instead of $persist_dir, causes this error:",
+        "   # [ERROR] [MY-010187] [Server] Could not open file '~/scoop/apps/mysql/current/data/host.err' for error logging: Permission denied",
+        "   Add-Content -Path \"$ini\" -Value \"[mysqld]`ndatadir=$($persist_dir -replace '\\\\', '/')/data\" -Encoding Ascii",
+        "   # Set username so it's not required when connecting",
+        "   Add-Content -Path \"$ini\" -Value \"[client]`nuser=root\" -Encoding Ascii",
+        "}"
+    ],
+    "post_install": "warn 'Database has been initialized (username: root, password: <blank>)'",
     "checkver": {
         "url": "https://dev.mysql.com/downloads/mysql/5.6.html",
         "re": "<h1>MySQL Community Server ([\\d.]+)"
@@ -52,6 +64,18 @@
                 "url": "https://dev.mysql.com/get/mysql-$version-win32.zip",
                 "extract_dir": "mysql-$version-win32"
             }
+        },
+        "hash": {
+            "url": "https://dev.mysql.com/downloads/mysql/5.6.html",
+            "regex": "(?sm)$basename.*?md5\">([A-Fa-f\\d]{32})"
         }
-    }
+    },
+    "notes": [
+        "Run 'mysqld --standalone' or 'mysqld --console' to start the Database,",
+        "or run following command as administrator to register MySQL as a service. See: https://dev.mysql.com/doc/refman/5.6/en/windows-start-service.html",
+        "",
+        "mysqld --install MySQL --defaults-file=\"$dir\\my.ini\"",
+        "",
+        "To stop and/or delete the Service run 'sc stop MySQL' and 'sc delete MySQL'."
+    ]
 }

--- a/bucket/mysql57.json
+++ b/bucket/mysql57.json
@@ -1,42 +1,62 @@
 {
-    "homepage": "https://dev.mysql.com/downloads/mysql/",
+    "homepage": "https://dev.mysql.com/downloads/mysql/5.7.html",
+    "description": "The world's most popular open-source database, supported by an active community.",
     "version": "5.7.29",
-    "license": "GPLv2",
+    "license": "GPL-2.0-only",
     "architecture": {
         "64bit": {
             "url": "https://dev.mysql.com/get/mysql-5.7.29-winx64.zip",
-            "hash": "864ac2695c5412733c848affeae0e8d7878e95122b7041728535ecf107ba02ca",
+            "hash": "md5:459e26ed1fdf08001851cf642338845e",
             "extract_dir": "mysql-5.7.29-winx64"
         },
         "32bit": {
             "url": "https://dev.mysql.com/get/mysql-5.7.29-win32.zip",
-            "hash": "0b3baf65c069ec94a6dcc813bc1c0cb6398abafd5a790b6f7f04c084f758d237",
+            "hash": "md5:000b342449a1742667cf33483546b3eb",
             "extract_dir": "mysql-5.7.29-win32"
         }
     },
     "bin": [
-        "bin\\mysqld.exe",
+        "bin\\innochecksum.exe",
+        "bin\\myisamchk.exe",
+        "bin\\myisamlog.exe",
+        "bin\\myisampack.exe",
+        "bin\\myisam_ftdump.exe",
         "bin\\mysql.exe",
-        "bin\\mysqldump.exe",
         "bin\\mysqladmin.exe",
         "bin\\mysqlbinlog.exe",
         "bin\\mysqlcheck.exe",
+        "bin\\mysqld.exe",
+        "bin\\mysqldump.exe",
         "bin\\mysqlimport.exe",
         "bin\\mysqlshow.exe",
         "bin\\mysqlslap.exe",
+        "bin\\mysql_config_editor.exe",
+        "bin\\mysql_secure_installation.exe",
+        "bin\\mysql_ssl_rsa_setup.exe",
+        "bin\\mysql_tzinfo_to_sql.exe",
+        "bin\\mysql_upgrade.exe",
         "bin\\my_print_defaults.exe"
     ],
+    "persist": [
+        "data",
+        "my.ini"
+    ],
+    "pre_install": [
+        "$ini = \"$dir\\my.ini\"",
+        "if (!(Test-Path \"$ini\")) {",
+        "   # Using $dir instead of $persist_dir, causes this error:",
+        "   # [ERROR] [MY-010187] [Server] Could not open file '~/scoop/apps/mysql/current/data/host.err' for error logging: Permission denied",
+        "   Add-Content -Path \"$ini\" -Value \"[mysqld]`ndatadir=$($persist_dir -replace '\\\\', '/')/data\" -Encoding Ascii",
+        "   # Set username so it's not required when connecting",
+        "   Add-Content -Path \"$ini\" -Value \"[client]`nuser=root\" -Encoding Ascii",
+        "}"
+    ],
     "post_install": [
-        "#Initialize data directory (without generating root password)",
-        "mysqld --initialize-insecure",
-        "",
-        "#Copy provided sample file to live file location",
-        "cp $dir/my-default.ini $dir/my.ini",
-        "",
-        "#Output client configuration to my.ini file so no username is required when connecting",
-        "echo \"\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append",
-        "echo \"[client]\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append",
-        "echo \"user=root\" | out-file \"$dir/my.ini\" -Encoding UTF8 -Append"
+        "if (!(Test-Path \"$dir\\data\\auto.cnf\")) {",
+        "    warn 'Initializing data directory ...'",
+        "    Invoke-ExternalCommand -FilePath \"$dir\\bin\\mysqld.exe\" -ArgumentList '--initialize-insecure'",
+        "    warn 'Database has been initialized (username: root, password: <blank>)'",
+        "}"
     ],
     "checkver": {
         "url": "https://dev.mysql.com/downloads/mysql/5.7.html",
@@ -52,6 +72,18 @@
                 "url": "https://dev.mysql.com/get/mysql-$version-win32.zip",
                 "extract_dir": "mysql-$version-win32"
             }
+        },
+        "hash": {
+            "url": "https://dev.mysql.com/downloads/mysql/5.7.html",
+            "regex": "(?sm)$basename.*?md5\">([A-Fa-f\\d]{32})"
         }
-    }
+    },
+    "notes": [
+        "Run 'mysqld --standalone' or 'mysqld --console' to start the Database,",
+        "or run following command as administrator to register MySQL as a service. See: https://dev.mysql.com/doc/refman/en/windows-start-service.html",
+        "",
+        "mysqld --install MySQL --defaults-file=\"$dir\\my.ini\"",
+        "",
+        "To stop and/or delete the Service run 'sc stop MySQL' and 'sc delete MySQL'."
+    ]
 }

--- a/bucket/mysql57.json
+++ b/bucket/mysql57.json
@@ -80,7 +80,7 @@
     },
     "notes": [
         "Run 'mysqld --standalone' or 'mysqld --console' to start the Database,",
-        "or run following command as administrator to register MySQL as a service. See: https://dev.mysql.com/doc/refman/en/windows-start-service.html",
+        "or run following command as administrator to register MySQL as a service. See: https://dev.mysql.com/doc/refman/5.7/en/windows-start-service.html",
         "",
         "mysqld --install MySQL --defaults-file=\"$dir\\my.ini\"",
         "",

